pkg/library/flatten/flatten.go

```diff
@@ -127,6 +123,9 @@ func recursiveResolve(workspace *dw.DevWorkspaceTemplateSpec, tooling ResolverTo
 		}
 	}
 
+	if err := mergeDevWorkspaceElements(resolvedContent, resolvedParent, pluginSpecContents...); err != nil {
+		return nil, fmt.Errorf("failed to merge DevWorkspace volumes: %w", err)
+	}
 	resolvedContent, err := overriding.MergeDevWorkspaceTemplateSpec(resolvedContent, resolvedParent, pluginSpecContents...)
```

Comment:
maybe let's move it there into pkg/library/flatten/merge.go as well?

Reply:
:+1: Good point. Squashed this change into the latest commit, PTAL